Tools/jit/template.c

```diff
@@ -91,8 +102,18 @@ _JIT_ENTRY(_PyInterpreterFrame *frame, PyObject **stack_pointer, PyThreadState *
     STACK_SHRINK(1);
 error_tier_two:
     _PyFrame_SetStackPointer(frame, stack_pointer);
+    tstate->previous_executor = (PyObject *)current_executor;
     return NULL;
 deoptimize:
     _PyFrame_SetStackPointer(frame, stack_pointer);
+    tstate->previous_executor = (PyObject *)current_executor;
     return _PyCode_CODE(_PyFrame_GetCode(frame)) + _target;
+side_exit:
+    {
+        _PyExitData *exit = &current_executor->exits[_target];
```

Comment:
I don't see this as a problem. If we change something in the interpreter, and forget to change it in the JIT, the tests will fail as everything crashes. It would be hard to miss.